services/madoc-ts/src/gateway/api.ts

```diff
@@ -391,6 +392,18 @@ export class ApiClient {
       : {};
   }
 
+  /// Search
+
+  async search(searchTerm: string, pageQuery?: number, facetType?: string, facetValue?: string) {
+    // Facet Types these are just one at a time for now, may switch to a post query with the json if a list!
+    return await this.request<SearchResponse>(
+      `/api/search/search?fulltext=${searchTerm}` +
+        `${pageQuery ? `&page=${pageQuery}` : ``}` +
+        `${facetType ? `&facet_type=${facetType}` : ``}` +
+        `${facetValue ? `&facet_subtype=${facetValue}` : ``}`
```

Comment:
We could use the `stringify()` function, that could make the query building a bit easier to maintain, `getCollections` below has an example.